src/lsm/forest.zig

```diff
@@ -1344,14 +1357,10 @@ fn CompactionPipelineType(comptime Forest: type, comptime Grid: type) type {
         }
 
         fn beat_grid_forfeit_all(self: *CompactionPipeline) void {
-            var i = self.compactions.count();
-            while (i > 0) {
-                i -= 1;
-
-                // We need to run this for all compactions that ran acquire - even if they
-                // transitioned to being finished, so we can't just use bar_active.
-                if (!self.beat_reserved.isSet(i)) continue;
-
+            // We need to run this for all compactions that ran acquire - even if they
+            // transitioned to being finished, so we can't just use bar_active.
+            var compactions_reserved = self.beat_reserved.iterator(.{});
+            while (compactions_reserved.next()) |i| {
```

Comment:
I didn't see any reason why we should forfeit in reverse order, but lmk if I've missed something.

Reply:
Yeah - it's not needed!